git-annex-remote-rclone

```diff
@@ -16,268 +14,175 @@
-				echo INITREMOTE-SUCCESS
-			else
-				echo INITREMOTE-FAILURE "Failed to create directory on remote. Ensure that 'rclone config' has been run."
-			fi
-		;;
-		PREPARE)
-			# Use GETCONFIG to get configuration settings,
-			# and do anything needed to get ready for using the
-			# special remote here.
-
-			getconfig prefix
-			REMOTE_PREFIX="$RET"
-
-			getconfig target
-			REMOTE_TARGET="$RET"
-
-			getconfig rclone_layout
-			RCLONE_LAYOUT="$RET"
-			validate_layout
-
-			echo PREPARE-SUCCESS
-		;;
-		TRANSFER)
-			op="$2"
-			key="$3"
-			shift 3
-			file="$*"
-			case "$op" in
-				STORE)
-					# Store the file to a location
-					# based on the key.
-					# XXX when at all possible, send PROGRESS
-					calclocation "$key"
-					if [ ! -e "$file" ]; then
-						echo TRANSFER-FAILURE STORE "$key" "asked to store non-existent file $file"
-					else
-						if runcmd rclone copy "$file" "$LOC"; then
-							echo TRANSFER-SUCCESS STORE "$key"
-						else
-							echo TRANSFER-FAILURE STORE "$key"
-						fi
-					fi
-				;;
-				RETRIEVE)
-					# Retrieve from a location based on
-					# the key, outputting to the file.
-					# XXX when easy to do, send PROGRESS
-					calclocation "$key"
-					# http://stackoverflow.com/questions/31396985/why-is-mktemp-on-os-x-broken-with-a-command-that-worked-on-linux
-					if GA_RC_TEMP_DIR=$(mktemp -d "${TMPDIR:-/tmp}/rclone-annex-tmp.XXXXXXXXX") &&
-						runcmd rclone copy "$LOC$key" "$GA_RC_TEMP_DIR" &&
-						mv "$GA_RC_TEMP_DIR/$key" "$file" &&
-						rmdir "$GA_RC_TEMP_DIR"; then
-						echo TRANSFER-SUCCESS RETRIEVE "$key"
-					else
-						echo TRANSFER-FAILURE RETRIEVE "$key"
-					fi
-				;;
-			esac
-		;;
-		CHECKPRESENT)
-			key="$2"
-			calclocation "$key"
-
-			do_checkpresent "$key" "$LOC$key"
-		;;
-		REMOVE)
-			key="$2"
-			calclocation "$key"
-
-			do_remove "$key" "$LOC$key"
-		;;
-		*)
-			# The requests listed above are all the ones
-			# that are required to be supported, so it's fine
-			# to say that any other request is unsupported.
-			echo UNSUPPORTED-REQUEST
-		;;
-	esac
-done
-
 
-# XXX anything that needs to be done at shutdown can be done here
+import os
+import re
+import sys
+import shutil
+import tempfile
+import subprocess
+
+def say(cmd, *messages):
+    print(cmd, *messages)
+    sys.stdout.flush()
+
+def ask(cmd, *query):
+    say(cmd, *query)
+    resp_key, _, resp_value = sys.stdin.readline().partition(" ")
+    if resp_key == "VALUE":
+        return resp_value.rstrip()
+    if resp_key == "CREDS":
+        return resp_value.rstrip().split(" ", 1)
+
+def rclone(*args):
+    proc = subprocess.Popen(("rclone",) + args, stderr=subprocess.STDOUT, stdout=subprocess.PIPE)
+    output = proc.stdout.read()
+    return proc.wait(), output.decode("utf8")
+
+def get_config(key):
+    return ask("GETCONFIG", key)
+
+def set_config(key, value):
+    say("SETCONFIG", key, value)
+
+def debug(*messages):
+    say("DEBUG", " ".join(map(str, messages)))
+
+
+class Remote(object):
```

Comment:
```suggestion
class Remote:
```

`(object)` as a base class is pointless in Python 3, and no one should be writing code for Python 2 in this day and age.

Reply:
> no one should be writing code for Python 2 in this day and age

I wish I could say that was an old habit. Vfx industry is a slow moving beast. Sigh...
Thanks for bringing it up though! Will change!